src/miner.cpp

```diff
@@ -298,8 +298,24 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn, CWallet *pwallet,
             ++nBlockTx;
             nBlockSigOps += nTxSigOps;
 
-            if (tx.type == NORMAL)
-                cnt++;
+            if (tx.type == NORMAL) {
+                BOOST_FOREACH(const CTxIn& txin, tx.vin) {
+                    TxInfo txinfo;
+                    if (!txinfo.init(txin.prevout)) {
+                        throw std::runtime_error("CreateNewBlock() : fetch input failed");
+                    }
+
+                    if (txinfo.GetTxOutColorOfIndex(txin.prevout.n) != TxFee.GetColor())
```

Comment:
`if (txinfo.GetTxOutColorOfIndex(txin.prevout.n) == TxFee.GetColor()) totalfee += txinfo.GetTxOutValueOfIndex(txin.prevout.n);` should be better?